timemachine/cpp/src/nonbonded_all_pairs.cu

```diff
@@ -365,14 +362,8 @@ void NonbondedAllPairs<RealType>::execute_device(
     // params are N, PARAMS_PER_ATOM
     // this needs to be an accumulated permute
     if (d_du_dp) {
-        // scattered assignment updates K_ <= N_ elements; the rest should be 0
-        gpuErrchk(cudaMemsetAsync(d_du_dp_buffer_, 0, N_ * PARAMS_PER_ATOM * sizeof(*d_du_dp_buffer_), stream));
-        k_scatter_assign<<<dim3(ceil_divide(K_, tpb), PARAMS_PER_ATOM, 1), tpb, 0, stream>>>(
-            K_, d_sorted_atom_idxs_, d_gathered_du_dp_, d_du_dp_buffer_);
-        gpuErrchk(cudaPeekAtLastError());
-
-        k_add_ull_to_ull<<<dim3(ceil_divide(N_, tpb), PARAMS_PER_ATOM, 1), tpb, 0, stream>>>(
-            N, d_du_dp_buffer_, d_du_dp);
+        k_scatter_accum<<<dim3(ceil_divide(K_, tpb), PARAMS_PER_ATOM, 1), tpb, 0, stream>>>(
```

Comment:
nit: I was confused at first that `k_scatter_accum` looks like it's intended to work with real types (given that the name of the template parameter is `RealType`). Maybe the parameter name could be changed to something more general, e.g. just `T`?

Reply:
Changed to T in [1a763be](https://github.com/proteneer/timemachine/pull/1049/commits/1a763becd7d76cb26d622a78b94a5ca6030e0079)